aptos-move/framework/aptos-framework/sources/configs/Stake.move

```diff
@@ -49,16 +49,20 @@ module AptosFramework::Stake {
         minimum_stake: u64,
         // maximum stakes allowed to join validator set
         maximum_stake: u64,
-        // should we store the ValidatorInfo in place?
-        validators: vector<address>,
-        last_update_time_secs: u64,
+        // active validators for the current epoch
+        active_validators: vector<ValidatorInfo>,
+        // pending validators to leave in next epoch (still active)
+        pending_inactive: vector<ValidatorInfo>,
+        // pending validators to join in next epoch
+        pending_active: vector<ValidatorInfo>,
+
     }
 
     /// Any user can delegate a stake.
-    fun delegate_stake(account: &signer, to: address, coin: Coin, locked_until_secs: u64) acquires ValidatorInfo, ValidatorSet {
+    fun delegate_stake(account: &signer, to: address, coin: Coin, locked_until_secs: u64) acquires StakePool, ValidatorSet {
```

Comment:
any measures to prevent account from using the delegated coins for other purposes?

Reply:
there's no way they can use the coins, it's wrapped in the Delegation struct. the permission is controlled by the functions defined for the structs.